### PR TITLE
Refactor the progress tracker to be non-asset-specific

### PR DESCRIPTION
**Pull request by volkangurel**
Refactor the progress tracker to be non-asset-specific and allow further refactoring of executable entrypoints

### CODE REVIEW

File: layer/main/asset.py
Comment (adrien-layer) [inline layer/main/asset.py:254]: other one is `mark_uploading` so maybe either this one is `mark_downloaded` or other one is `mark_asset_uploading`?

**Review by adrien-layer — APPROVED** [review]
minor comment

**Review by cihati — COMMENTED** [review]
I think @aleksmitov should review this before we merge

File: layer/main/asset.py
Comment (volkangurel) [reply layer/main/asset.py:254]: good point, done!

**Review by aleksmitov — APPROVED** [review]
Looks good to me, just please make exhaustive enough manual tests to ensure there are no UI regressions.

**Comment by volkangurel** [comment]
Thanks @aleksmitov, already done!

**Pull request merged**
